src/portal-impl.c

```diff
@@ -270,6 +304,119 @@ load_installed_portals (gboolean opt_verbose)
   implementations = g_list_sort (implementations, sort_impl_by_use_in_and_name);
 }
 
+void
+load_portal_configuration (gboolean opt_verbose)
+{
+  g_autoptr(PortalConfig) conf = NULL;
+  const char **desktops;
+  const char *portal_dir;
+
+  /* We need to override this in the tests */
+  portal_dir = g_getenv ("XDG_DESKTOP_PORTAL_DIR");
+  if (portal_dir == NULL)
+    portal_dir = DATADIR "/xdg-desktop-portal";
+
+  conf = g_new0 (PortalConfig, 1);
+
+  desktops = get_current_lowercase_desktops ();
+  for (size_t i = 0; desktops[i] != NULL; i++)
+    {
+      g_autofree char *portals_conf = g_strdup_printf ("%s-portals.conf", desktops[i]);
+      g_autofree char *path = g_build_filename (portal_dir, portals_conf, NULL);
+      g_auto(GStrv) ifaces;
+
+      g_autoptr(GKeyFile) key_file = g_key_file_new ();
+
+      g_debug ("Looking for portals configuration in '%s'", path);
+      if (!g_key_file_load_from_file (key_file, path, G_KEY_FILE_NONE, NULL))
+        continue;
+
+      if (opt_verbose)
+        g_debug ("Loading portal configuration file '%s' for desktop '%s'",
+                 path,
+                 desktops[i]);
+
+      ifaces = g_key_file_get_keys (key_file, "preferred", NULL, NULL);
+
+      g_autoptr(GPtrArray) interfaces = g_ptr_array_new_full (g_strv_length (ifaces) + 1, NULL);
+
+      if (ifaces != NULL)
+        {
+          for (size_t i = 0; ifaces[i] != NULL; i++)
```

Comment:
Instead of nesting an if and a for loop, I'd rather have `for (size_t i = 0; ifaces != NULL && ifaces[i] != NULL; i++)`

Reply:
Once you hoist the variable declarations, the condition becomes useful to avoid allocating stuff pointlessly.